src/main/java/org/jboss/ejb/client/EJBRootContext.java

```diff
@@ -83,9 +83,15 @@ protected Object lookupNative(final Name name) throws NamingException {
             }
         }
         if (beanName == null) {
+            if (size == 3) {
+                // name is of the form appName/moduleName/distinctName
+                return new RelativeContext(new FastHashtable<>(getEnvironment()), this, SimpleName.of(name));
+            }
```

Comment:
The Javadoc in the legacy EjbJndiNameParser indicates that a name of the form a/b/c will be considered to be a partial app/module/distinct lookup as opposed to a lookup of a bean name without a view. See https://github.com/jbossas/jboss-ejb-client/blob/3.x/src/main/java/org/jboss/ejb/client/naming/ejb/EjbJndiNameParser.java#L40-L42.